src/interface/serial_node.py

```diff
@@ -229,7 +229,7 @@ def read_firmware_timestamp(self):
 
 def discover(idxOffset, config, isS32BPillFlag=False, *args, **kwargs):
     nodes = []
-    config_ser_ports = getattr(config, 'SERIAL_PORTS', [])
+    config_ser_ports = config.get_item('GENERAL', 'SERIAL_PORTS')
```

Comment:
This no longer has the empty-list default value, so will break if 'SERIAL_PORTS' is not present.

Reply:
SERIAL_PORTS is always present in the internal default configuration that `config.json` applies on top of..

Reply:
Still better to have a default value; there's no guarantee of that assumption at this point in the code